File: frontend/src/api/homework.ts
import request from '@/utils/request'

export interface HomeworkForm {
  title: string
  content?: string
  course_id?: string
  grade_id?: string
  class_id?: string
  homework_type?: string
  total_score?: number
  submit_start?: string
  submit_end?: string
  notify_enabled?: boolean
}

export interface SubmissionForm {
  content?: string
  attachment_urls?: string[]
}

export interface WrongQuestionForm {
  question_content: string
  question_type?: string
  correct_answer?: string
  student_answer?: string
  score?: number
  source_type?: string
  source_id?: string
}

export interface FeedbackForm {
  feedback_type: string
  content: string
}

export function getHomeworks(params?: { course_id?: string; class_id?: string; status?: string; page?: number; page_size?: number }) {
  return request.get('/homework/homeworks', { params })
}

export function createHomework(data: HomeworkForm) {
  return request.post('/homework/homeworks', data)
}

export function updateHomework(id: string, data: Partial<HomeworkForm>) {
  return request.put(`/homework/homeworks/${id}`, data)
}

export function getSubmissions(homeworkId: string) {
  return request.get(`/homework/homeworks/${homeworkId}/submissions`)
}

export function submitHomework(homeworkId: string, data: SubmissionForm) {
  return request.post(`/homework/homeworks/${homeworkId}/submit`, data)
}

export function gradeSubmission(submissionId: string, score: number, feedback?: string) {
  return request.put(`/homework/submissions/${submissionId}/grade`, null, { params: { score, feedback } })
}

export function getWrongQuestions(params?: { student_id?: string; is_mastered?: boolean; page?: number; page_size?: number }) {
  return request.get('/homework/wrong-questions', { params })
}

export function createWrongQuestion(data: WrongQuestionForm) {
  return request.post('/homework/wrong-questions', data)
}

export function getFeedbacks(homeworkId: string) {
  return request.get(`/homework/homeworks/${homeworkId}/feedbacks`)
}

export function createFeedback(homeworkId: string, data: FeedbackForm) {
  return request.post(`/homework/homeworks/${homeworkId}/feedback`, data)
}

export function getHomeworkStats() {
  return request.get('/homework/homeworks/stats')
}

File: backend/app/schemas/homework.py
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class HomeworkCreate(BaseModel):
    title: str
    content: Optional[str] = None
    course_id: Optional[str] = None
    grade_id: Optional[str] = None
    class_ids: Optional[list] = None
    homework_type: str = "online"
    total_score: float = 100
    submit_start: Optional[datetime] = None
    submit_end: Optional[datetime] = None
    attachments: Optional[list] = None
    notify_enabled: bool = True


class HomeworkUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    homework_type: Optional[str] = None
    total_score: Optional[float] = None
    submit_start: Optional[datetime] = None
    submit_end: Optional[datetime] = None
    attachments: Optional[list] = None
    status: Optional[str] = None


class HomeworkSubmissionCreate(BaseModel):
    content: Optional[str] = None
    attachment_urls: Optional[list] = None


class HomeworkSubmissionUpdate(BaseModel):
    score: Optional[float] = None
    feedback: Optional[str] = None


class WrongQuestionCreate(BaseModel):
    question_content: str
    question_type: Optional[str] = None
    correct_answer: Optional[str] = None
    student_answer: Optional[str] = None
    score: Optional[float] = None
    source_type: str = "exam"
    source_id: Optional[str] = None


class HomeworkFeedbackCreate(BaseModel):
    feedback_type: str
    content: str


class HomeworkFeedbackReply(BaseModel):
    content: str